lighthouse-core/lib/errors.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const strings = require('./strings');
+
+class LighthouseError extends Error {
+  constructor(errorDefinition) {
+    super(errorDefinition.code);
+    Error.captureStackTrace(this, LighthouseError);
+    this.friendlyMessage = errorDefinition.message;
```

Comment:
And we can show your provided message in the CLI instead of https://github.com/GoogleChrome/lighthouse/blob/b433a2a5329fb9bb35bbc8802fb16b95c27355a2/lighthouse-cli/run.js#L85

Reply:
done, cleaned that area up a bit too